rm/rm-node-updater/src/main/java/org/ow2/proactive/resourcemanager/updater/RMNodeUpdater.java

```diff
@@ -46,113 +74,496 @@
-                            System.out.println("Releasing the lock on " + lockFile.getAbsoluteFile());
-                            lock.release();
-                            channel.close();
+    @Override
+    protected String fillParameters(final CommandLine cl, final Options options) {
+        String parentResult = super.fillParameters(cl, options);
 
-                            return false;
-                        }
-                    }
+        if (cl.hasOption(OPTION_NODE_URL)) {
+            nodeJarUrl = cl.getOptionValue(OPTION_NODE_URL);
+        } else if (System.getProperty(NODE_URL_PROPERTY) != null) {
+            nodeJarUrl = System.getProperty(NODE_URL_PROPERTY);
+        } else {
+            logger.error("Option " + OPTION_NODE_JAR_URL_NAME + " must be specified or java property " +
+                         NODE_URL_PROPERTY + " must be set");
+            System.exit(ExitStatus.RMNODE_PARSE_ERROR.exitCode);
+        }
 
-                    FileUtils.copyURLToFile(new URL(jarUrl), destination);
-                    System.out.println("Download finished");
+        if (cl.hasOption(OPTION_NODE_SAVEAS)) {
+            nodeJarSaveAs = cl.getOptionValue(OPTION_NODE_SAVEAS);
+        } else if (System.getProperty(NODE_JAR_SAVEAS_PROPERTY) != null) {
+            nodeJarSaveAs = System.getProperty(NODE_JAR_SAVEAS_PROPERTY);
+        } else {
+            logger.error("Option " + OPTION_NODE_JAR_SAVE_AS_NAME + " must be specified or java property " +
+                         NODE_JAR_SAVEAS_PROPERTY + " must be set");
+            System.exit(ExitStatus.RMNODE_PARSE_ERROR.exitCode);
+        }
 
-                    if (lock != null && lockFile != null) {
-                        System.out.println("Releasing the lock on " + lockFile.getAbsoluteFile());
-                        lock.release();
+        if (cl.hasOption(OPTION_NODE_AUTOMATIC)) {
+            automaticRelaunch = true;
+        } else if (System.getProperty(NODE_UPDATER_AUTOMATIC_RELAUNCH) != null) {
+            automaticRelaunch = "true".equals(System.getProperty(NODE_UPDATER_AUTOMATIC_RELAUNCH));
+        }
+        return parentResult;
+    }
+
+    @Override
+    protected String parseCommandLine(String[] args) {
+        final Options options = new Options();
+
+        fillOptions(options);
+
+        final CommandLineParser parser = new DefaultParser();
+
+        CommandLine cl;
+        try {
+            cl = parser.parse(options, args);
+            //now we update this object's fields given the options.
+            String nodeName = fillParameters(cl, options);
+            //check the user supplied values
+            //performed after fillParameters to be able to override fillParameters in subclasses
+            checkUserSuppliedParameters();
+            return nodeName;
+        } catch (ParseException pe) {
+            pe.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)